http/src/v0/refs.rs

```diff
@@ -0,0 +1,76 @@
+use futures::stream;
+use ipfs::error::Error;
+use ipfs::{Ipfs, IpfsTypes};
+use serde::Serialize;
+use warp::hyper::Body;
+use warp::{path, Filter, Rejection, Reply};
+
+use super::support::with_ipfs;
+
+#[derive(Serialize, Debug)]
+struct RefsResponseItem {
+    #[serde(rename = "Err")]
+    err: String,
+
+    #[serde(rename = "Ref")]
+    refs: String,
+}
+
+/// Handling of https://docs-beta.ipfs.io/reference/http/api/#api-v0-refs-local
+pub fn local<T: IpfsTypes>(
+    ipfs: &Ipfs<T>,
+) -> impl Filter<Extract = impl Reply, Error = Rejection> + Clone {
+    path!("refs" / "local")
+        .and(with_ipfs(ipfs))
+        .and_then(inner_local)
+}
+
+async fn inner_local<T: IpfsTypes>(ipfs: Ipfs<T>) -> Result<impl Reply, Rejection> {
+    let refs: Vec<Result<String, Error>> = ipfs
+        .refs_local()
+        .await
+        .unwrap()
+        .iter()
+        // TODO: Error handling
+        // .map_err(|err| RefsResponseItem { refs: "", err: "" })
+        .map(|cid| cid.as_ref().unwrap().to_string())
+        .map(|refs| RefsResponseItem {
+            refs,
+            err: "".to_string(),
+        })
+        .map(|response| serde_json::to_string(&response))
+        .map(|ref_string| Ok(format!("{}{}", ref_string.unwrap(), "\n")))
```

Comment:
It doesn't quite make sense for the json serialization to fail here. In case I had to debug such I'd prefer if the panic message would say something more relevant, like `ref_string.expect("Unexpected failure to serialize into json")`..

Note that `ref_string` is misleading variable name: the `ref_string` is not the `{ "Ref": "--> this value right here <--", ... }` but the whole json serialized, wrapped in a `Result<String, serde_json::Error>`, so maybe it could be `json` instead?